### ISSUE TITLE
Trace kafka client (producer) does not utilize message keys

### ISSUE THREAD

**Issue by slysterous** [beatlabs/patron#85]
## Is your feature request related to a problem? Please describe

TL;DR
Message key is hard-coded nil, this disables the utilisation of Kafka message keys.

Let's analyse an example, i have 2 services that communicate through a topic.

The first service produces multiple requests to the topic.
The second service consumes the requests and acts upon them.

In this particular case it is important that the order of the requests is maintained upon consumption.

In order to achieve that goal, message keys must be utilised so that the producer can place the message on a partition based on that key, (in order to be able to enforce message order on a partition level) .

## Is this a bug? Please provide steps to reproduce, a failing test etc.

This is not a bug, it is a feature request in the form of an enhancement.

## Describe the solution

The createProducerMessage() on kafka.go receives a pointer on a Message struct.
That Message struct should include a key property that could optionally be filled with a key.
That key will be passed on the sarama.ProducerMessage key property.
